src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.cs

```diff
@@ -77,6 +77,61 @@ public object Invoke(string method, params object?[] args)
             return res;
         }
 
+        public struct EventListenerOptions {
+            public bool Capture;
+            public bool Once;
+            public bool Passive;
+            public object? Signal;
+        }
+
+        public int AddEventListener(string name, Delegate listener, EventListenerOptions? options = null)
+        {
+            var optionsDict = options.HasValue
+                ? new JSObject()
+                : null;
+
+            try {
+                if (options?.Signal != null)
+                    throw new NotImplementedException("EventListenerOptions.Signal");
+
+                var jsfunc = Runtime.GetWeakDelegateHandle(listener);
```

Comment:
If we could wrap the `listener` with our own lambda here and call `ReleaseInFlight(arg1)` we would stop leaking inflight JSObjects of the event argument (which I introduced recently).

Reply:
I will look into this, great idea